### PR TITLE
[C#] Fix copyright, company and authors metadata in assembly and nuget.

### PR DESCRIPTION
**Pull request by eerhardt**
Follow up from #3891 to fix up references to `The Apache Software Foundation` in the C# assembly and NuGet metadata.

I also updated the `Copyright` to match 

https://github.com/apache/arrow/blob/686c44ca6472a86f6908debc21330270c4c01c90/NOTICE.txt#L1-L2

@wesm @kou

### CODE REVIEW

File: csharp/Directory.Build.props [hunk 1]
```diff
@@ -10,8 +10,8 @@
   <!-- AssemblyInfo properties -->
   <PropertyGroup>
     <Product>Apache Arrow library</Product>
-    <Copyright>2018 Apache Software Foundation</Copyright>
-    <Company>Apache</Company>
+    <Copyright>Copyright 2016 The Apache Software Foundation</Copyright>
```
Comment (pitrou) [inline csharp/Directory.Build.props]: The "2016" looks weird, but I'm not a lawyer. Either "2019" or "2016-2019" would look more expected. @wesm Any opinion?

**Review by pitrou — COMMENTED** [review]
Looks fine on the principle. Just one question.

File: csharp/Directory.Build.props (hunk 1 above)
Comment (eerhardt) [reply csharp/Directory.Build.props]: I took this from https://github.com/apache/arrow/blob/master/NOTICE.txt#L2. I can change it to whatever folks want.

Comment (wesm) [reply csharp/Directory.Build.props]: If we have to have a year then 2016-2019 would be it

Comment (eerhardt) [reply csharp/Directory.Build.props]: I can drop the year too, if you want...

Comment (pitrou) [reply csharp/Directory.Build.props]: Better fix it to "2016-2019", IMHO. This is what https://www.apache.org/legal/src-headers.html#notice seems to recommend.

Comment (eerhardt) [reply csharp/Directory.Build.props]: OK, fixed to 

`<Copyright>Copyright 2016-2019 The Apache Software Foundation</Copyright>`

**Review by kou — APPROVED** [review]
+1
I think that we can merge this after CI is passed.

**Comment by pitrou** [comment]
@eerhardt You may as well fix the copyright years in the NOTICE file while you're at it.

**Review by pitrou — APPROVED** [review]
+1

**Pull request merged**